geopandas/tests/test_pandas_methods.py

```diff
@@ -603,6 +603,12 @@ def test_apply_preserves_geom_col_name(df):
     assert result.geometry.name == "geom"
 
 
+def test_df_apply_returning_series(df):
+    # https://github.com/geopandas/geopandas/issues/2283
+    result = df.apply(lambda row: row.geometry, axis=1)
+    assert_geoseries_equal(result, df.geometry, check_crs=False)
```

Comment:
```suggestion
    result = df.apply(lambda row: row.geometry, axis=1)
    assert_geoseries_equal(result, df.geometry, check_crs=False)
    
    result = df.apply(lambda row: row.value1, axis=1)
    assert_series_equal(result, df["value1"])
```

Probably not needed to cover the specific regression, but I thought it would be good to cover both cases of the result being a normal Series vs GeoSeries